Use color picker for frame colors

naglepuff opened the pull request:
This PR makes the interface for changing frame colors more user friendly by adding a color swatch to the input field. This swatch is clickable, and brings up a color picker when clicked.

naglepuff committed:
Use color picker for frame colors

naglepuff committed:
Use `TiffFrame` type

annehaley reviewed (commented):
The code looks good, but in testing it out, I found that the channels didn't update with the new colors. Additionally, when I select a color on the edge of the color picker (dragging and releasing my mouse outside of the picker to do so, as shown in the video below), the whole dialog disappears. Is there a way you can modify the off-click of the dialog to only fire if the _start_ of the click is off the dialog?

**Edit:** I see the update button now, but is it possible to fire the update when a color is done being selected (i.e. when the singular picker is unfocused)?

https://user-images.githubusercontent.com/44912689/165541795-d9646d02-0346-4935-97e3-15acab28d4fe.mov

naglepuff committed:
Use dialog for editing frames

There were some issues with using `v-menu` components. Namely,
clicking on the color picker and dragging outside would cause
the frame menu to close.

naglepuff commented:
The channels only update when the `Update` button is clicked. This change was made in #142.

To address your usability concern, check out f2300f5. I did some digging and couldn't find anything specific to the layout I had before, but I found some [similar issues](https://github.com/vuetifyjs/vuetify/issues/12461) with `v-dialog`s. Those issues were addressed in [this PR](https://github.com/vuetifyjs/vuetify/pull/13244), so I changed our UI a bit to get the desired functionality.

annehaley reviewed (approved):
That works much better and makes it more clear that the user needs to hit the update button. Thanks!
